pkg/querier/distributor_queryable_streaming.go

```diff
@@ -12,17 +12,21 @@ import (
 	"github.com/grafana/mimir/pkg/storage/series"
 )
 
+type streamingChunkSeriesConfig struct {
```

Comment:
Don't love this name, open to other suggestions.

Reply:
`streamingChunkSeriesContext`? 

I'm a bigger fan of smaller packages, so I'd also move all the `streamingChunkSeries` structs in their own package (`stream`?) and drop the prefix from the names. So this struct becomes `context` or `config`. But most packages in mimir are quite large, so this may not fly.

Reply:
> `streamingChunkSeriesContext`?

I like it, I'll open a PR to rename this.

> I'm a bigger fan of smaller packages, so I'd also move all the `streamingChunkSeries` structs in their own package (`stream`?) and drop the prefix from the names. So this struct becomes `context` or `config`. But most packages in mimir are quite large, so this may not fly.

Unfortunately this would create a import loop: the `querier` package would reference the `stream` package for `streamingChunkSeries`, but `streamingChunkSeriesConfig` in the `stream` package references `chunkIteratorFunc` from the `querier` package.